ProcessOptimizer/space/space.py

```diff
@@ -1,917 +1,896 @@
+    def __init__(self, low, high, transform=None, name=None):
+        """Search space dimension that can take on integer values.
+
+        Parameters
+        ----------
+        * `low` [int]:
+            Lower bound (inclusive).
+
+        * `high` [int]:
+            Upper bound (inclusive).
+
+        * `transform` ["identity", "normalize", optional]:
+            The following transformations are supported.
+
+            - "identity", (default) the transformed space is the same as the
+              original space.
+            - "normalize", the transformed space is scaled to be between
+              0 and 1.
+
+        * `name` [str or None]:
+            Name associated with dimension, e.g., "number of trees".
+        """
+        if high <= low:
+            raise ValueError("the lower bound {} has to be less than the"
+                             " upper bound {}".format(low, high))
+        self.low = low
+        self.high = high
+        self.name = name
+
+        if transform is None:
+            transform = "identity"
+
+        self.transform_ = transform
+
+        if transform not in ["normalize", "identity"]:
+            raise ValueError("transform should be 'normalize' or 'identity'"
+                             " got {}".format(self.transform_))
+        if transform == "normalize":
+            self.transformer = Normalize(low, high, is_int=True)
+        else:
+            self.transformer = Identity()
+
+    def __eq__(self, other):
+        return (type(self) is type(other) and
+                np.allclose([self.low], [other.low]) and
+                np.allclose([self.high], [other.high]))
+
+    def __repr__(self):
+        return "Integer(low={}, high={})".format(self.low, self.high)
+
+    def inverse_transform(self, Xt):
+        """Inverse transform samples from the warped space back into the
+           orignal space.
+        """
+        # The concatenation of all transformed dimensions makes Xt to be
+        # of type float, hence the required cast back to int.
+        return super(Integer, self).inverse_transform(Xt).astype(np.int64)
+
+    @property
+    def bounds(self):
+        return (self.low, self.high)
+
+    def __contains__(self, point):
+        return self.low <= point <= self.high
+
+    @property
+    def transformed_bounds(self):
+        if self.transform_ == "normalize":
+            return 0, 1
+        else:
+            return (self.low, self.high)
+
+    def distance(self, a, b):
+        """Compute distance between point `a` and `b`.
+
+        Parameters
+        ----------
+        * `a` [int]
+            First point.
+
+        * `b` [int]
+            Second point.
+        """
+        if not (a in self and b in self):
+            raise RuntimeError("Can only compute distance for values within "
+                               "the space, not %s and %s." % (a, b))
+        return abs(a - b)
+
+    def _sample(self, point_list=Iterable[float]) -> np.ndarray:
```

Comment:
Is the `=` for `point_list` intentional? I assume this is meant to be a type hint, in which case the line should be 
```python
def _sample(self, point_list: Iterable[float]) -> np.ndarray:
```